src/main/java/org/candlepin/subscriptions/tally/SnapshotSummaryProducer.java

```diff
@@ -63,30 +80,44 @@ protected SnapshotSummaryProducer(
     this.summaryMapper = summaryMapper;
   }
 
-  public void produceTallySummaryMessages(Map<String, List<TallySnapshot>> newAndUpdatedSnapshots) {
-    AtomicInteger totalTallies = new AtomicInteger();
-    newAndUpdatedSnapshots.forEach(
-        (orgId, snapshots) ->
-            /* Filter snapshots, as we only deal with hourly, non Any fields
-            and measurement types other than Total
-            when we transmit the tally summary message to the BillableUsage component. */
-            snapshots.stream()
-                .filter(SnapshotSummaryProducer::filterByHourlyAndNotAnySnapshots)
-                .map(
-                    snapshot -> {
-                      removeTotalMeasurements(snapshot);
-                      return snapshot;
-                    })
-                .sorted(Comparator.comparing(TallySnapshot::getSnapshotDate))
-                .map(snapshot -> summaryMapper.mapSnapshots(orgId, List.of(snapshot)))
+  public void produceTallySummaryMessages(
+      Map<String, List<TallySnapshot>> newAndUpdatedSnapshots,
+      List<Granularity> granularities,
+      Predicate<TallySnapshot> filter) {
+    Map<Granularity, Map<String, List<TallySnapshot>>> groupedSnapshots =
+        groupByGranularity(newAndUpdatedSnapshots, granularities);
+    granularities.forEach(
+        granularity -> {
+          AtomicInteger totalTallies = new AtomicInteger();
+          if (groupedSnapshots.get(granularity) != null) {
+            groupedSnapshots
+                .get(granularity)
                 .forEach(
-                    summary -> {
-                      kafkaRetryTemplate.execute(
-                          ctx -> tallySummaryKafkaTemplate.send(tallySummaryTopic, orgId, summary));
-                      totalTallies.getAndIncrement();
-                    }));
+                    (orgId, snapshots) ->
+                        /* Filter snapshots for specific granularity, non-Any fields
+                        and measurement types other than Total
+                        when we transmit the tally summary message to the BillableUsage component. */
+                        snapshots.stream()
+                            .filter(filter)
+                            .map(
+                                snapshot -> {
+                                  removeTotalMeasurements(snapshot);
```

Comment:
@wottop this is what's causing the test_tally_rhel_addons_variants failures. The getTallyReport api depends on their being 'TOTAL' measurements on the DAILY snapshots. This removal also affects the actual snapshot db entity.